### PR TITLE
To Do list: add & remove

### PR DESCRIPTION
**Pull request by rudicarrilloypr**
### Project requirements
- Remove all hardcoded items from the tasks array.
- Create a new JavaScript file for the new functionality.
- Implement a function for adding a new task (add a new element to the array).
- Implement a function for deleting a task (remove an element from the array).
- Implement a function for editing task descriptions.
- By default new tasks should have the property `completed` set to `false` and the property `index` set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5).
- Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should set to 1)..
- All changes to the To Do List should be saved in local storage.

### CODE REVIEW

**Comment by rudicarrilloypr** [comment]
Webhint linter not working correctly, Microverse let us know to submit our project, and code reviewers will ignore this linter issue, the rest of the linters and projector requirements work as they should.

File: src/taskFunctions.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    completed: false,
+    index: highestIndex + 1,
+  };
+
+  tasks.push(newTask);
+
+  setTasks(tasks);
+
+  taskInput.value = '';
+  populateTodoList(tasks, document.getElementById('todo-list'));
```
Comment (mikethreels) [inline src/taskFunctions.js]: - [x] according to the project requirements All changes to the To Do List should be saved in local storage. which is not the case at the moment.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Rudi, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Good job creating the crud methods.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/taskFunctions.js [hunk 2]
```diff
@@ -7,19 +7,30 @@ export function getTasks() {
 
 export function setTasks(newTasks) {
   tasks = newTasks;
+  // eslint-disable-next-line no-use-before-define
+  saveTasksToLocalStorage(tasks);
+}
+
+export function saveTasksToLocalStorage(tasks) {
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+}
```
Comment (omarsalem7) [inline src/taskFunctions.js:16]: - [x] As stated by the previous reviewer, kindly consider after saving todos in local storage and refresh the page the todos displayed from local storage.

**Review by omarsalem7 — CHANGES_REQUESTED** [review]
Hi @rudicarrilloypr 😊, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

Almost near

 ## Highlights
✔️ The Pull Request has a proper title and description
✔️ The data set in local storage 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @rudicarrilloypr,

✔️ Books reappear on page reload

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**